### ISSUE TITLE
[LAP Rescue Run v2] LS0.05 runtime-equivalent seed42 with exact epoch continuation

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#68]
## Objective

Authorize one new **epoch-0** LAP-GNN seed42 validation-only LS0.05 scientific execution after the independently reviewed technical invalidation of Issue #60.

Issue #60 remains permanently invalid and is not interpreted. Its checkpoint/history MUST NOT be loaded or used.

Scientific question is unchanged from Issue #60:

> Does training-only label smoothing with epsilon `0.05` materially reduce the frozen LAP-GNN train-validation macro-F1 gap while preserving or improving validation accuracy/macro-F1 under the same scientific training protocol?

The only execution amendment is the independently proven runtime-equivalent lifecycle plus exact completed-epoch continuation protection from Issues #61/#63.

## Exact execution source

Run only from exact merged main commit:

`157c8a87f84e0aa53b4432393b99a3a560c0974e`

This main includes:
- accepted LS0.05 implementation PR #59;
- runtime-equivalent PR #62;
- exact-continuation PR #64;
- unrelated accepted WS-HPG implementation, which must not affect LAP scientific payload.

Candidate module:

`research/candidates/tf_lap_gnn_ls005_runtime_equivalent/`

Locked scientific/frozen identities:
- frozen scientific payload SHA-256 `286be711a53b76511bcf3b9bf949fad694f7c7d272392f9defc56f4914822c0e`
- execution contract SHA-256 `14acc2750875a25922007459161a137158d8040805e616166be923f63658bf22`
- frozen trainer SHA-256 `4c3cb1aa311578038ff656cb7d119103ae5a651135f8ee1c76e37c2c04c1fc75`
- frozen validation-only wrapper SHA-256 `c94c122066fdd19210c8ba64a2a61567b249fad4f69c69cb4236b68cce6ff7b4`
- LS0.05 loss adapter SHA-256 `72435f59cbc138f4e2184b384d0a4e2c85cdd0082655a6371e0a61ee6362e1e5`
- lean config SHA-256 `8c1947e86eba24cb9c783d6a3c1e11cd5c4d4dcce4db0e84e10f617ae7c8d38f`
- continuation capsule contract SHA-256 `4d689f29a1866ff60a74cbc168b75727aac05670f75cec61bff165a62c1c02e1`
- continuation.py SHA-256 `05e0ed233f3eeda4e2a99b7b4df696b97fb9735cdf2b9661322d8a81802e81c6`
- continuation_equivalence.py SHA-256 `418f6a049a44e51d6f31c628bebc8d37346f81628d73dd9851b10b5ff6c3e998`
- trainer_continuation_adapter.py SHA-256 `8c188846adab119fefebf661155404f34a04a4ce589dd9c6d8b3615e6b0da7c2`
- final continuation-enabled train_validation_only.py SHA-256 `ec0bf836a890312dbd68ed183f7cf5d31cdc99d90b2fa6f8a1ad17e613dc3905`
- model parameters/trainable variables `1,061,192 / 127`
- dataset signature `fer2013_train28709_val3589_test3589`.

Any mismatch => STOP before training.

## Scientific intervention and frozen protocol

Training target only:

`target = (1 - 0.05) * one_hot(y, 7) + 0.05 / 7`

All scientific quantities remain Issue #60 frozen:
- seed 42
- train 28,709
- validation 3,589
- test forbidden
- batch 16
- eval batch 32
- graph workers 2
- tf-data prefetch 2
- tf-data parallel calls 1
- graph cache 64
- mixed precision ON
- XLA OFF
- memory growth ON
- Torch-compatible AdamW
- LR 3e-4
- weight decay 1e-3
- no gradient clipping
- max epochs 90
- ReduceLROnPlateau on hard-CE val_loss, factor 0.5, patience 5, threshold 1e-4, min LR 3e-5
- early stopping on hard-CE val_loss, minimum epoch 30, patience 15
- earliest strict maximum val_accuracy checkpoint selection
- train-only prior corruption schedule unchanged 0.1 -> 0.2 -> 0.3
- validation hard CE unchanged
- final clean-train hard CE unchanged.

Runtime-equivalent lifecycle change already proven exact:
- per-epoch clean-train evaluation removed;
- validation remains every epoch;
- after training, selected checkpoint is reloaded `compile=False`;
- full clean-train hard CE exactly once;
- full validation hard CE exactly once.

No scientific hyperparameter changed.

## Registered runtime

Require exact baseline-compatible Kaggle class before training:
- Linux
- Python 3.12.12
- TensorFlow 2.18.1
- Keras 3.15.0
- CUDA 12.5.1
- cuDNN major 9
- exactly 2 x Tesla T4
- no operation-determinism change.

Any mismatch => preflight stop, attempt not consumed.

## Inputs

Train CSV:
`/kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv`

MediaPipe prior root:
`/kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue`

Clean graph cache:
`/kaggle/input/datasets/irthn1311/ofix7-mid-seed42-records`

Never read/hash/open `test.csv` or test sample records.

## First-run exact command

Start from epoch 0 only. Continuation root must be fresh/empty.

```bash
python -m research.candidates.tf_lap_gnn_ls005_runtime_equivalent.train_validation_only \
  --config research/candidates/tf_lap_gnn_ls005_runtime_equivalent/configs/fer2013_ofix7_mid_seed42_ls005_lean.yaml \
  --fer-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --prior-root /kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue \
  --clean-graph-cache-dir /kaggle/input/datasets/irthn1311/ofix7-mid-seed42-records \
  --output-root /kaggle/working/tf_lap_gnn_ls005_rescue_seed42_lean \
  --continuation-root /kaggle/working/tf_lap_gnn_ls005_rescue_seed42_lean_continuation \
  --device gpu \
  --graph-workers 2 \
  --tf-data-prefetch 2 \
  --tf-data-parallel-calls 1 \
  --graph-cache-size 64 \
  --batch-size 16 \
  --eval-batch-size 32 \
  --mixed-precision \
  --no-xla \
  --memory-growth \
  --no-resume
```

No `--limit-*`. No `--resume-capsule-root` on first launch.

Once this first epoch-0 training command starts, the scientific attempt is consumed.

## External-interruption continuation rule

If and only if the platform externally terminates the run before normal/registered early-stop completion:

- do not interpret partial performance;
- preserve the latest complete continuation lineage;
- a later process may continue only from the most recent verified `LATEST.json` capsule;
- never resume a partial epoch;
- never use Issue #60 artifacts;
- every capsule/member/scientific identity must verify exactly;
- continuation is part of this same scientific attempt because exact fresh-process equivalence was preregistered and proven at tolerance 0.0.

For a continuation process, use a **fresh output root and fresh continuation write root**, supplying the preserved previous continuation lineage via `--resume-capsule-root`. Do not silently fall back to an older capsule.

If capsule verification fails, scientific result is invalid and no automatic new attempt is authorized.

## Frozen comparator

Primary LAP selected-checkpoint comparator (epoch 31):
- val accuracy `0.6319308999721371`
- val macro-F1 `0.5938407974340496`
- clean-train macro-F1 `0.8207655611897143`
- macro gap `22.69247637556647 pp`.

Secondary trajectory-only best macro reference:
- best val macro `0.601166548701511` at epoch 26
- clean-train macro `0.7562805286580438`
- gap `15.511397995653287 pp`.

No test comparator.

## Primary endpoint and decision classifier

Use selected checkpoint = earliest strict maximum validation accuracy.

Report selected checkpoint train/val accuracy, macro-F1 and hard-CE loss, gaps, epoch, and:
- delta val accuracy pp vs `0.6319308999721371`
- delta val macro pp vs `0.5938407974340496`
- macro-gap improvement vs `22.69247637556647 pp`.

Decision precedence is copied unchanged from Issue #60:

1. `LAP_LS005_STRETCH_RESCUE`
2. `LAP_LS005_RESCUE_PASS`
3. `LAP_LS005_GENERALIZATION_RESCUE_WITH_SMALL_ACCURACY_COST`
4. `LAP_LS005_OVERREGULARIZED`
5. `LAP_LS005_NO_CLEAR_RESCUE`
6. `LAP_LS005_REGRESSION`
7. `LAP_LS005_INCONCLUSIVE`

### STRETCH_RESCUE
All:
- val accuracy >= 0.6500
- val macro >= 0.6200
- macro gap <= 8.0 pp.

### RESCUE_PASS
All:
- val accuracy >= 0.6319308999721371
- val macro >= 0.5938407974340496
- macro gap <= 10.0 pp.

### GENERALIZATION_RESCUE_WITH_SMALL_ACCURACY_COST
Unless higher priority applies, all:
- val accuracy >= 0.6219308999721371
- macro-gap improvement >= 7.5 pp
- val macro >= 0.5838407974340496.

### OVERREGULARIZED
Unless higher priority applies:
- macro-gap improvement >= 7.5 pp
- AND val accuracy change < -1.0 pp OR val macro change < -1.0 pp.

### NO_CLEAR_RESCUE
Unless higher priority applies:
- abs val-accuracy change < 1.0 pp
- macro-gap improvement < 7.5 pp
- abs val-macro change <= 1.0 pp.

### REGRESSION
Unless higher priority applies:
- no qualifying gap rescue
- AND val accuracy <= baseline -1.0 pp OR val macro <= baseline -1.0 pp.

Else INCONCLUSIVE.

## Validity gates

Scientific decision valid only if:
- exact checkout and all hashes PASS;
- registered runtime PASS;
- subprocess ultimately completes successfully, including any exact continuation segments;
- complete lineage has no gaps/corruption;
- frozen completion marker valid;
- lean completion marker valid;
- training_validation_completed=true;
- final_test_skipped=true;
- test_accessed=false;
- Issue60 artifacts used=false;
- LS training binding restored;
- evaluator hard CE unchanged;
- model identity exact;
- selected checkpoint metadata agrees with earliest strict global max val accuracy across complete reconstructed history;
- train/val history internally complete;
- no forbidden test artifacts/access.

If any fails: `scientific_result_valid=false`, `scientific_interpretation=null`.

## Evidence

Capture complete stdout/stderr and pre-run identities. Preserve all output roots and continuation lineages. After valid completion, package evidence once and report ZIP SHA-256/bytes/member count/integrity, embedded `.keras` SHA-256/member count/integrity, forbidden test/CSV scan, and `test_access=false`.

Do not repack after final hash.

Final valid status:

`LAP_GNN_LS005_RESCUE_SEED42_LEAN_COMPLETE_PENDING_INDEPENDENT_REVIEW`

Technical/invalid status:

`LAP_GNN_LS005_RESCUE_SEED42_LEAN_INVALID_PENDING_INDEPENDENT_REVIEW`

## Interpretation boundary

A positive result supports only that LS0.05 improves the fit/generalization trade-off under the frozen protocol. It does not prove label noise is the unique cause of LAP overfit.

No outcome authorizes test access or epsilon sweep.
